fenix/app/src/main/java/org/mozilla/fenix/onboarding/WidgetPinnedReceiver.kt

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding
+
+import android.app.PendingIntent
+import android.appwidget.AppWidgetManager
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.localbroadcastmanager.content.LocalBroadcastManager
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import mozilla.components.support.utils.PendingIntentUtils
+
+/**
+ * Receiver required to catch callback from Launcher when prompted
+ * to add search widget from the Juno Onboarding
+ */
+class WidgetPinnedReceiver : BroadcastReceiver() {
+
+    companion object {
+        const val ACTION = "org.mozilla.fenix.onboarding.WidgetPinnedReceiver.PIN_SEARCH_WIDGET_SUCCESS"
+
+        /**
+         * Prepare success callback for when requesting to pin Search Widget
+         */
+        fun getPendingIntent(context: Context): PendingIntent {
+            val callbackIntent = Intent(context, WidgetPinnedReceiver::class.java)
+            val bundle = Bundle()
+            bundle.putInt(AppWidgetManager.EXTRA_APPWIDGET_ID, 1)
+            callbackIntent.putExtras(bundle)
+            return PendingIntent.getBroadcast(
+                context,
+                0,
+                callbackIntent,
+                PendingIntentUtils.defaultFlags or PendingIntent.FLAG_UPDATE_CURRENT,
+            )
+        }
+    }
+
+    /**
+     * Object containing boolean that updates behavior of Add Search Widget
+     * card from JunoOnboardingScreen
+     */
+    object WidgetPinnedBooleanState {
+        private val _value = MutableStateFlow(false)
+        val value: StateFlow<Boolean> = _value
+
+        /**
+         * Update state when resumed to add search widget card
+         * and the widget was added successfully
+         */
+        fun updateBoolean(newValue: Boolean) {
```

Comment:
I wonder if the param is necessary? Could perhaps just be something such as `widgetWasPinned`